### DISCUSSION TITLE
During git push - Unable to access Fatal Error

### DISCUSSION THREAD

**Discussion by Sumanshu-Nankana** [community/community#23487] [New to GitHub]
<p>While running git push origin master<br>
I am getting below error.</p>
<p>remote: Permission to Sumanshu-Nankana/Docker.git denied to Sumanshu-Nankana.<br>
fatal: unable to access ‘<a href="https://github.com/Sumanshu-Nankana/Docker.git/" rel="noopener nofollow ugc">https://github.com/Sumanshu-Nankana/Docker.git/</a>’: The requested URL returned error: 403</p>
<ol>
<li>Password and Username is correct (as it’s working on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>)</li>
<li>I tried with Public Access Token (because I have 2 Factor Auth enable)</li>
<li>I am using https (so no ssh key concept)</li>
<li>I am able to push using VS CODE , but unable to push using Terminal (getting this error first time - probably after enabling 2Factor Auth)</li>
</ol>
<p>But with both password and token, I am getting same ‘Unable to access’, with token (I get some extra message as well i.e. The requested URL returned error: 403)</p>
<p>can anyone help me, what is wrong. Though I am</p>

**Comment by strafe** [comment, marked as answer]
<p>Did you make sure to give the PAT the necessary permissions when creating it?</p>

**Comment by Feri982** [comment]
<p>Hi a have same problem, unable to access when I use git push at command line. What PAT is? How can I give that permission (if that’s is the problem) after? Can I? Please help me, thanks</p>

**Comment by airtower-luna** [comment]
<p>“PAT” is short for “personal access token”. The token is a long and complicated string that you can use similarly to a password. Here’s the documentation on how to create one: <a href="https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token" class="inline-onebox">Creating a personal access token - GitHub Docs</a></p>
<p>If you already have PATs, you can see them on the <strong>Personal access tokens</strong> page in your settings. Click the one you want to edit and you should see (among other things) a list of scopes and a button to update the token below.</p>

**Comment by Feri982** [comment]
<p>Thanks for the reply. I will try. Can I use that token as a password at command line?</p>
<ol start="2021">
<li>augusztus 14., szombat dátummal Airtower via GitHub Support Community &lt;<a href="mailto:githubcommunity@discoursemail.com">githubcommunity@discoursemail.com</a>&gt; ezt írta:</li>
</ol>

**Comment by airtower-luna** [comment]
<p>Yes, see <a href="https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token#using-a-token-on-the-command-line">Using a token on the command line</a>. That also has a link to how to cache the token (so you don’t have to enter it for every operation, only once in a while).</p>
<p>Usually it’s a good idea to store the token in a password manage. After all it purposefully isn’t easy to remember. :wink:</p>

**Comment by Feri982** [comment]
<p>Hi I still can’t log in, I made the token I got message<br>
Hey Feri982!</p>
<p>A personal access token (Feri) with admin:repo_hook, delete_repo, and repo scopes was recently added to your account.<br>
I’m confused, I used that long token as password and I tried with my old password I tried many options but I can see that as on picture. Attached</p>
<ol start="2021">
<li>augusztus 14., szombat dátummal Airtower via GitHub Support Community &lt;<a href="mailto:githubcommunity@discoursemail.com">githubcommunity@discoursemail.com</a>&gt; ezt írta:</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/0/a0edabd466dbc311cc4b32b758560c7a068b7bce.jpeg" data-download-href="/uploads/short-url/mXDBpODbgagqBzVBQN3Y4kPetFI.jpeg?dl=1" title="16289445404867252739230410586220.jpg" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/88894594/181077361-0cd8e3b5-96ae-40e2-b9ed-b4d8ca5f6a23.jpeg" alt="16289445404867252739230410586220.jpg" data-base62-sha1="mXDBpODbgagqBzVBQN3Y4kPetFI" width="375" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/0/a0edabd466dbc311cc4b32b758560c7a068b7bce_2_375x500.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/a/0/a0edabd466dbc311cc4b32b758560c7a068b7bce_2_562x750.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/0/a0edabd466dbc311cc4b32b758560c7a068b7bce_2_750x1000.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/0/a0edabd466dbc311cc4b32b758560c7a068b7bce_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">16289445404867252739230410586220.jpg</span><span class="informations">3456×4608 3.08 MB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>It looks like the token is visible at the top of your photo. This forum is public, so you have shared the token with the whole world. You need to revoke the token <strong>now</strong> and generate a new one. :warning:</p>
<p>Other than that, it looks like you typed <code>Username: Feri982</code> instead of <code>Feri982</code> for your user name.</p>

**Comment by Feri982** [comment]
<p>okay i changed my token. other than i did not. when i used on prompt line hat command : git push “my git link”<br>
Username: Feri982<br>
Password: <em>i hide my token</em><br>
so after that i got that screen. now i upload to youtube how i did , if i did wrong please tell me which step is wrong. if anybody can send me step by step basic video how i need to do, i m confused.<br>
here is the, and sorry for my bad English :slight_smile:<br>
<a href="https://youtu.be/e7Ig2t74ys8" rel="noopener nofollow ugc">https://youtu.be/e7Ig2t74ys8</a></p>
<p>Huawei &lt;<a href="mailto:82feri@gmail.com">82feri@gmail.com</a>&gt; ezt írta (időpont: 2021. aug. 14., Szo, 17:59):</p>

**Comment by airtower-luna** [comment]
<p>I’m not sure what that that window is, maybe some GitHub integration that comes with Git for Windows? On Linux I get the command line prompt immediately.</p>
<p>What you do at the command line prompt looks correct in the video. In particular, unlike in the screenshot the password prompt now says <code>Password for 'https://Feri982@github.com'</code>. The part between <code>https://</code> and <code>@</code> needs to be your user name.</p>
<p>I just tried with my own username and PAT and it worked, so probably something goes wrong with copying the PAT. In the video it looks and sounds like you select and copy the token text. That can easily go wrong, for example if you accidentally copy a space next to the token, too, it won’t work.</p>
<p>There’s a little “clipboard” icon next to the token when you generate it: <img src="https://user-images.githubusercontent.com/5814919/181077363-2aa026e9-0b95-499d-9d2e-8574cf5921cf.png" alt="grafik" data-base62-sha1="5y3PGpgTkaVhz3SLDf2zdkcu2N5" width="27" height="25"> Try clicking that instead to copy the token.</p>

**Comment by Feri982** [comment]
<p>Hi I did like copy paste with mouse and I did what you say tap the clipboard icon and paste.</p>

**Comment by airtower-luna** [comment]
<p>And it still doesn’t work? That’s really odd! In that case I can only give the same advice as here, unfortunately:</p><aside class="quote" data-post="4" data-topic="195566">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/token-not-working-in-command-line/195566/4">Token not working in command line</a> <a class="badge-wrapper  bullet" href="/c/github-help/48"><span class="badge-category-bg" style="background-color: #6a737d;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Need help with your GitHub account or the core features of GitHub? This is the place to look for answers.">GitHub Help</span></a>
  </div>
  <blockquote>
    That’s odd! In that case the only thing I can think of is to share your platform (OS) and Git version (ideally check for updates if you haven’t already), maybe someone else has encountered a similar issue. 
Or try using SSH instead of HTTPS.
  </blockquote>
</aside>

**Comment by Feri982** [comment]
<p>hi unfortunately that is not works, same as the other user what you sent me the link. how can i try with ssh? im very new i dont understand lots of things. how can i change https to ssh?</p>

**Comment by Feri982** [comment]
<p>now succeded! i deleted my .git files from my pc, reinstalled the git program, make everything again, and now working. i just had 1 file to add and commit so wasnt a big deal. cheers my friend</p>

**Comment by somixyz** [comment]
<p>Did you try ?</p>
<p>git push -f origin </p>

**Comment by airtower-luna** [comment]
<p><code>git push -f</code> is a potentially destructive operation. :warning:</p>
<p>There are times when it’s needed, but if you just “try” it in case of a problem you risk losing data. And it definitely won’t help with an authentication issue.</p>
